src/Diagnostics.zig

```diff
@@ -30,6 +31,22 @@ pub const Message = struct {
             actual: u21,
             resembles: u21,
         },
+        attr_arg_count: struct {
+            attribute: Attribute.Tag,
+            expected: u32,
+        },
+        attr_arg_type: struct {
+            expected: Attribute.ArgumentType,
+            actual: Attribute.ArgumentType,
+        },
+        attr_enum: struct {
```

Comment:
This union field is too big - I think it would be tricky to format the string at the callsite, so how would you feel about excluding it from the error message? We would still retain the correct caret position e.g. the difference between:

```
./test.c:1:25: error: Unknown `mode` argument 'bad'. Possible values are: 'byte', '__byte__', 'word', '__word__', 'pointer', '__pointer__'
int __attribute__((mode(bad))) foo;
                        ^
```

and

```
./test.c:1:25: error: Unknown `mode` argument. Possible values are: 'byte', '__byte__', 'word', '__word__', 'pointer', '__pointer__'
int __attribute__((mode(bad))) foo;
                        ^
```

Reply:
Fine by me, Clang/GCC only give the name of the bad argument. Also why are the normalized values also listed as enum members?

Reply:
You're right, the double underscore versions should not be in the enum and `diagnoseIdent` should normalize the name. There are a bunch of other values that should be in the enum: https://gcc.gnu.org/onlinedocs/gccint/Machine-Modes.html (for some reason that page lists them as e.g. `QImode ` but the actual value is `QI`)